src/documents/PromisesAPI.md

```diff
@@ -0,0 +1,236 @@
+
+### fluid.promise.fireTransformEvent(event, payload[, options])
+
+* `event {[Event](InfusionEventSystem.md)}` A "pseudoevent" whose listeners are to be treated as sucessive (asynchronous) stages in the process of transforming a payload.
+* `payload {Any}` The original payload input to the transforming chain.
+* `options {Object}` [optional] A set of additional options to be supplied to each listener in the transform chain. Accepts two special options:
+    * `reverse: {Boolean}` If `true`, the sequence of handlers will be notified in reverse order
+    * `filterNamespaces: {Array of String}` A collection of event [namespaces](InfusionEventSystem.md#namespaced-listeners) to be filtered out of the processing chain for this particular firing 
+
+This is a slightly esoteric but very powerful API. To get a sense of its overall function, it could be compared with the standard [pipeline](https://github.com/cujojs/when/blob/master/docs/api.md#whenpipeline)
+algorithm supplied with [when.js](https://github.com/cujojs/when) - the concept is that an "initial payload" (which may be empty) is successively
+transformed by sequentially, possibly asynchronous, stages of a pipeline of functions. Each function accepts the return value of its predecessor,
+and may synchronously return a transformed payload, or a promise asynchronously yielding such a payload. It may also of course also return a promise which rejects,
+terminating the transform chain.
+
+This packaging of the pipeline algorithm is significantly more powerful, since it can call upon the [priority](Priorities.md) feature of 
+standard Infusion [events](InfusionEventSystem.md) in order to allow processing elements to be integrated together from multiple sources,
+with each one free to insert themselves at any symbolically identified (by means of `before:` and `after:` type constraint priorities) position
+in the chain.
+
+Each listener to the "transform event" (we call this a "pseudoevent" precisely because each listener does not receive the *same* argument list
+as with traditional events, but instead receives the returned and resolved value of its precessor) has the following signature:
+
+* `listener {Function:(previousValue {Any}, options {Object}) →  Any|Promise}` where `previousValue` is the resolved return from the previous listener notified in the chain,
+or the initial `payload` value supplied to `fluid.promise.fireTransformEvent` if it is the first in the chain, and `options` is the last argument to `fluid.promise.fireTransformEvent`.
+
+## Inverse API recognised by promises consumed by sequential algorithms
+
+Both `fluid.promise.sequence` and `fluid.promise.fireTransformEvent` will recognise the following method supplied by the user on any promise
+returned by one of sources in the sequence:
+
+### promise.accumulateRejectionReason(error)
+
+* `error: {Object|Error}` A rejection which has been received from a promise "to the right` of this one in a promise sequence.
```

Comment:
The point is that "before" and "after" is ambiguous with respect to whether the sequence algorithm is travelling from left to right or not.